Sources/RoktUXHelper/Data/Model/RoktPluginViewState.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  RoktPluginViewState.swift
+//  Pods
+//
+//  Copyright 2020 Rokt Pte Ltd
+//
+//  Licensed under the Rokt Software Development Kit (SDK) Terms of Use
+//  Version 2.0 (the "License");
+//
+//  You may not use this file except in compliance with the License.
+//
+//  You may obtain a copy of the License at https://rokt.com/sdk-license-2-0/
+
+import Foundation
+
+@objc public class RoktPluginViewState: NSObject {
+    public let pluginId: String
+    public var offerIndex: Int
+    public var isPluginDismissed: Bool
+    public var customStateMap: CustomStateMap?
+    
+    /// Shortcut initialiser that, given a pluginId, defaults to standard, initial plugin view states
+    public convenience init(pluginId: String) {
+        self.init(pluginId: pluginId,
+                  offerIndex: 0,
+                  isPluginDismissed: false,
+                  customStateMap: nil)
+    }
+
+    public init(pluginId: String,
+                offerIndex: Int,
+                isPluginDismissed: Bool,
+                customStateMap: CustomStateMap?) {
+        self.pluginId = pluginId
+        self.offerIndex = offerIndex
+        self.isPluginDismissed = isPluginDismissed
+        self.customStateMap = customStateMap
+    }
+    public override func isEqual(_ object: Any?) -> Bool {
+        guard let rhs = object as? RoktPluginViewState else { return false }
+
+        return (self.pluginId == rhs.pluginId &&
+                self.offerIndex == rhs.offerIndex &&
+                self.isPluginDismissed == rhs.isPluginDismissed &&
+                self.customStateMap == rhs.customStateMap)
+    }
+}
+
+@objc public class RoktPluginViewStateUpdates: NSObject {
```

Comment:
whats the reason we are not reusing `RoktPluginViewState`?

Reply:
to ensure RoktPluginViewState doesn't allow for nils (everything should be set to the view state values or to defaults).
but then the RoktPluginViewStateUpdates that will get sent in the callback needs to allow for nils so we can populate it with the existing cached view state properties before being written out to the cache file.
is there a better way to do this 🤔

Reply:
makes sense, because the risk is low (invalid object would just fetch from api) and it's in the caller's interest to supply a valid payload. i would reuse the same type.

Reply:
ok thanks understood the mission 🫡 will revisit this in the next PR when i add caching of sent events